### PR TITLE
Create 0000-yarn-global-run

### PR DESCRIPTION
**Pull request by Haroenv**
see also https://github.com/yarnpkg/yarn/issues/4392

[rendered](https://github.com/Haroenv/rfcs/blob/patch-1/accepted/0000-yarn-global-run.md)

### CODE REVIEW

**Comment by bestander** [comment]
Is it similar to `npx`?

**Comment by Haroenv** [comment]
It definitely is

File: accepted/0000-yarn-global-run [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Npx already exists and fills this position for now. 
+
+Adding a flag like `--exact` to `yarn create` so that the `create-` prefix isn't 
+automatically added
+
+# Unresolved questions
+
+- will the installed package be installed globally
+- is the name good
```
Comment (BYK) [inline accepted/0000-yarn-global-run]: I think we should add an item about a possible consolidation with `yarn create`

File: accepted/0000-yarn-global-run [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Alternatives
+
+Npx already exists and fills this position for now. 
+
+Adding a flag like `--exact` to `yarn create` so that the `create-` prefix isn't 
+automatically added
+
+# Unresolved questions
+
+- will the installed package be installed globally
```
Comment (BYK) [inline accepted/0000-yarn-global-run]: `yarn create` installs it globally and doesn't clean it up after the run for future invocations. I expect this to behave the same unless people object.

File: accepted/0000-yarn-global-run [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Drawbacks
+
+It can cause confusion with regard to `yarn create`, since it has a very similar
+goal. 
+
+# Alternatives
+
+Npx already exists and fills this position for now. 
+
+Adding a flag like `--exact` to `yarn create` so that the `create-` prefix isn't
```
Comment (BYK) [inline accepted/0000-yarn-global-run]: Can you think of more? May be a separate binary like `yr`?

File: accepted/0000-yarn-global-run [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+`create-`. 
+
+# How We Teach This
+
+Yarn run and Yarn global already exist, so this is not introducing new concepts, 
+especially since `yarn create` already exists.
+
+# Drawbacks
+
+It can cause confusion with regard to `yarn create`, since it has a very similar
```
Comment (BYK) [inline accepted/0000-yarn-global-run]: May be we can come up with some safety measure for `yarn create` that limits damage/impact for scripts adhering to certain standards and `yarn global run` becomes the "escape hatch" everything else?

One thing that I can think of is not making binaries from `yarn create` installs globally available at all so you __have to__ run `yarn create react-app` even after installing it.

File: accepted/0000-yarn-global-run [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+# Detailed design
+
+The design would be exactly the same as `yarn create`, but without the prefixing of
+`create-`. 
+
+# How We Teach This
+
+Yarn run and Yarn global already exist, so this is not introducing new concepts, 
+especially since `yarn create` already exists.
```
Comment (BYK) [inline accepted/0000-yarn-global-run]: We may wanna differentiate it a bit more. Also, a blog post would probably be helpful along with docs.

**Comment by BYK** [comment]
I think this is an interesting and good idea. What do you think @yarnpkg/core

**Comment by thejameskyle** [comment]
I'm not sure about the name `run`. When I first read `yarn global run` I expected it to work exactly like the `yarn run` command.

File: accepted/0000-yarn-global-run [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Some packages that are similar in design to `create-react-app` ([vue-cli](https://yarn.pm/vue-cli),
+[preact-cli](https://yarn.pm/preact-cli), [@angular/cli](https://yarn.pm/@angular/cli)
+...) don't have a name that's prefixed with `create-`, and thus prevent the usage of
+`yarn create`, even though it would be perfectly suitable for it. 
+
+# Detailed design
+
+The design would be exactly the same as `yarn create`, but without the prefixing of
+`create-`.
```
Comment (thejameskyle) [inline accepted/0000-yarn-global-run]: Could this replace `yarn create`?

Comment (Haroenv) [reply accepted/0000-yarn-global-run]: I guess that's a option

**Comment by ljharb** [comment]
npx is nice because it's x like "exec".

**Comment by ljharb** [comment]
(to clarify https://github.com/yarnpkg/rfcs/pull/81#issuecomment-329871642, i was suggesting that the name be `yarn exec`)

**Comment by bestander** [comment]
I'll be boring and suggest considering other options like a separate package or promote an alias in bash for that.
Less features, more time we can spend on maintaining others :)

**Comment by Haroenv** [comment]
this is long superseded by yarn dlx

**Comment by jerrygreen** [comment]
I tried `yarn dlx` but that doesn't seem to work

**Pull request closed without merging**